src/ceph.rs

```diff
@@ -59,10 +59,11 @@ pub enum CephCommandTypes {
 pub(crate) fn get_error(n: c_int) -> RadosResult<String> {
     let mut buf = vec![0u8; 256];
     unsafe {
-        strerror_r(n, buf.as_mut_ptr() as *mut ::libc::c_char, buf.len());
-        let message = String::from_utf8_lossy(&buf).into_owned();
-        Ok(message)
+        strerror_r(n, buf.as_mut_ptr() as *mut c_char, buf.len());
     }
+    let num_bytes = buf.iter().position(|x| x==&0u8);
+    buf.truncate(num_bytes.unwrap_or(0));
```

Comment:
Instead of two steps while not do this in the iterator itself?

example: https://play.rust-lang.org/?gist=88c09b3318607629bff17157612378c3&version=stable

Reply:
This allocates a 3rd buffer and I was trying to avoid that if possible.  I keep buf in place, truncate it to the proper length and then allocate another heap buffer for the string.

Reply:
Ah, indeed; why not try:
```
buf.retain(|&x| x != 0u8);
```